### PR TITLE
We shouldn't write to re-created tables for 2 mins

### PR DESCRIPTION
**Pull request by sb2nov**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

R: @chamikaramj PTAL

### CODE REVIEW

File: sdks/python/apache_beam/io/gcp/bigquery.py [hunk 1]
```diff
@@ -1002,12 +1002,23 @@ def get_or_create_table(
...
       # if write_disposition == BigQueryDisposition.WRITE_TRUNCATE we delete
       # the table before this point.
-      return self._create_table(project_id=project_id,
-                                dataset_id=dataset_id,
-                                table_id=table_id,
-                                schema=schema or found_table.schema)
+      if write_disposition == BigQueryDisposition.WRITE_TRUNCATE:
+        # BigQuery can route data to the old table for 2 mins max so wait
+        # that much time before creating the table and writing it
+        logging.info('Sleeping for 150 seconds before the write as BigQuery ' +
```
Comment (chamikaramj) [inline sdks/python/apache_beam/io/gcp/bigquery.py]: logging.warn

File: sdks/python/apache_beam/io/gcp/bigquery.py [hunk 2]
```diff
@@ -1002,12 +1002,23 @@ def get_or_create_table(
...
+                                         table_id=table_id,
+                                         schema=schema or found_table.schema)
       # if write_disposition == BigQueryDisposition.WRITE_TRUNCATE we delete
       # the table before this point.
-      return self._create_table(project_id=project_id,
-                                dataset_id=dataset_id,
-                                table_id=table_id,
-                                schema=schema or found_table.schema)
+      if write_disposition == BigQueryDisposition.WRITE_TRUNCATE:
+        # BigQuery can route data to the old table for 2 mins max so wait
```
Comment (chamikaramj) [inline sdks/python/apache_beam/io/gcp/bigquery.py]: Is there a link from BQ that we can add here ?

File: sdks/python/apache_beam/io/gcp/bigquery.py (hunk 1 above)
Comment (sb2nov) [inline sdks/python/apache_beam/io/gcp/bigquery.py]: Done.

File: sdks/python/apache_beam/io/gcp/bigquery.py (hunk 2 above)
Comment (sb2nov) [inline sdks/python/apache_beam/io/gcp/bigquery.py]: I couldn't find anything in Google Searching for it.

**Comment by chamikaramj** [comment]
Retest this please

**Comment by sb2nov** [comment]
Finally passed. Should be good to merge.

**Comment by chamikaramj** [comment]
LGTM. Merging.

**Pull request closed without merging**